File: ToyLanguage6/src/Model/MyIDictionary.java
package Model;

import java.util.Map;
import java.util.Set;

public interface MyIDictionary<K,V> {
    boolean isDefined(K id);

    void update(K id, V val);

    void add(K id, V val);

    V lookup(K id) throws MyException;

    int size();

    void remove(K key);

    boolean isEmpty();

    V get(K key);

    Map<K,V> getContent();

    Set<K> getKeys();
}

File: ToyLanguage7/ToyLanguage7/MyIDictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ToyLanguage7
{
    interface MyIDictionary<K,V>
    {
        bool isDefined(K id);
        void update(K id, V val);
        void add(K id, V val);
        V lookup(K id);
        int size();
        void remove(K key);
        bool isEmpty();
        V get(K key);

    }
}
